### DISCUSSION TITLE
Copilot Business : Where is the input data stored when i ask a question using Copilot?

### DISCUSSION THREAD

**Discussion by istaica** [community/community#70214] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hello,

I am using Copilot for Business and i would like to understand how the input data is process by Copilot when i ask a question or select a code snippet, and if it is stored temporarily somewhere. 

Thanks,
Ioana

**Comment by FilipJaskovic** [comment]
GitHub Copilot for Business utilizes data from file content and additional sources to improve its service. It collects User Engagement Data to measure metrics like latency and feature engagement. Prompts and Suggestions, generated during interaction, are sent in real-time but not retained. The collected data is used by GitHub and Microsoft for service provision, algorithm optimization, and user behavior analysis. Users can manage data collection, retention, and processing, yet some data remains essential for service use and will continue being shared with Microsoft. https://docs.github.com/en/enterprise-cloud@latest/copilot/overview-of-github-copilot/about-github-copilot-for-business

**Reply by FilipJaskovic** [reply, marked as answer]
To clarify: 

When you use Copilot for Business and input data like asking a question or selecting a code snippet, your interactions are processed in real-time to generate Suggestions. The data, referred to as Prompts, is sent to GitHub Copilot’s servers only in real-time and is not retained. Similarly, the Suggestions generated based on your Prompts are transmitted back to you in real-time and are also not retained. The data collected, such as User Engagement Data, helps in measuring certain metrics and improving the service, but is stored in a manner that doesn’t directly identify you .
